File: lab2/lab3.cpp
#include <iostream>
using namespace std;

int main() {
    int x;
    int steps = 0;
    cout << "Please enter a positive integer greater than 1: ";
    cin >> x;

    cout << x;
    while (x != 1) {
        if (x % 2 == 0) {
            x /= 2;
        } else {
            x = x * 3 + 1;
        }
        cout << " -> " << x;
        steps++;
    }
    cout << "\nTotal steps: " << steps << endl;


    int n;
    while (true) {
        cout << "Please enter a number between 10 and 100: ";
        cin >> n;
        if (n >= 10 && n <= 100) break;
        cout << "Invalid input. ";
    }

    int fizz = 0, buzz = 0, fizzbuzz = 0;

    for (int i = 1; i <= n; i++) {

        if (i % 7 == 0) {
            cout << "(" << i << " is skipped)" << endl;
            continue;
        }
        if (i % 3 == 0 && i % 5 == 0) {
            cout << "FizzBuzz" << endl;
            fizzbuzz++;
        } else if (i % 3 == 0) {
            cout << "Fizz" << endl;
            fizz++;
        } else if (i % 5 == 0) {
            cout << "Buzz" << endl;
            buzz++;
        } else {
            cout << i << endl;
        }
    }

    cout << "Summary" << endl;
    cout << "Fizz count: " << fizz << endl;
    cout << "Buzz count: " << buzz << endl;
    cout << "FizzBuzz count: " << fizzbuzz << endl;



        int a;

        while (true) {
            std::cout << "Please enter a number between 3 and 9: ";
            std::cin >> a;
            if (a >= 3 && a <= 9) {
                break;
            }
            std::cout << "Invalid input. ";
        }

        for (int i = 1; i <= 2 * a - 1; i++) {
            int k = a - std::abs(a - i);
            for (int j = 0; j < k; j++) {
                std::cout << "*";
            }
            std::cout << std::endl;
        }

        return 0;
    }

File: lab2/lab3.py
n = int(input("Please enter a positive integer greater than 1: "))
steps = 0
print(n, end="")

while n != 1:
    if n % 2 == 0:
        n = n // 2
    else:
        n = n * 3 + 1
    print(f" -> {n}", end="")
    steps += 1

print(f"\nTotal steps: {steps}")

while True:
    n = int(input("Please enter a number between 10 and 100: "))
    if 10 <= n <= 100:
        break
    print("Invalid input.", end=" ")

fizz_count = 0
buzz_count = 0
fizz_buzz_count = 0

for i in range(1, n + 1):
    if i % 7 == 0:
        print(f"({i} is skipped)" )
        continue

    if i % 3 == 0 and i % 5 == 0:
        print("FizzBuzz")
        fizz_buzz_count += 1
    elif i % 3 == 0:
        print("Fizz")
        fizz_count += 1
    elif i % 5 == 0:
        print("Buzz")
        buzz_count += 1
    else:
        print(i)

print("Summary")
print(f"Fizz count: {fizz_count}")
print(f"Buzz count: {buzz_count}")
print(f"FizzBuzz count: {fizz_buzz_count}")



while True:
    n = int(input("Please enter a number between 3 and 9: "))
    if 3 <= n <= 9:
        break
    print("Invalid input.", end=" ")

for i in range(1, 2 * n):
    k = n - abs(n - i)
    print("*" * k)